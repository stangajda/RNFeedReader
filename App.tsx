/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React, {useEffect, useState} from 'react';
import {ActivityIndicator, FlatList, Image, SafeAreaView, Text, View, StyleSheet} from 'react-native';

type Movie = {
  id: string;
  poster_path: string;
  title: string;
  vote_average: number;
  vote_count: number;
};

function App(): JSX.Element {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState<Movie[]>([]);

  const getMovies = async () => {
    try {
      const response = await fetch('https://api.themoviedb.org/3/trending/movie/day?api_key=babcada8d42a5fd4857231c42240debd');
      const json = await response.json();
      setData(json.results);
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    getMovies();
  }, []);

  return (
    <SafeAreaView style={styles.safeAreaView}>
      <View style={styles.container}>
        {isLoading ? (
          <ActivityIndicator />
        ) : (
          <FlatList
            data={data}
            keyExtractor={({id}) => id}
            renderItem={({item}) => (
              <View style={styles.listItem}>
                <Image source={{uri: `https://image.tmdb.org/t/p/w200${item.poster_path}`}} style={styles.image} />
                <View style={{flexDirection: "column", flex: 1 }}>
                  <Text numberOfLines={2} ellipsizeMode="tail" style={styles.title}>
                    {item.title}
                  </Text>
                  <View style={styles.ratingContainer}>
                    <Text style={styles.rating}>{item.vote_average} </Text> 
                    <Text>({item.vote_count})</Text>
                  </View>
                </View>
              </View>
            )}
          />
        )}
      </View>
    </SafeAreaView>
)};

const styles = StyleSheet.create({
  safeAreaView: {
    flex: 1
  },
  container: {
    padding: 24
  },
  listItem: {
    flexDirection: "row",
    padding: 8
  },
  image: {
    width: 64,
    height: 88,
    marginRight: 8
  },
  title: {
    flexShrink: 1,
    fontSize: 16,
    fontWeight: 'bold'
  },
  ratingContainer: {
    flexDirection: "row",
    flex: 1,
    alignItems: "center"
  },
  rating: {
    fontWeight: 'bold'
  }
});

export default App;
