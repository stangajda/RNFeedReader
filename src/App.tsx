import React from 'react';
import {ActivityIndicator, SafeAreaView, View, Text} from 'react-native';
import Styles from './Components/styles';
import MovieList from './Components/MovieList';
import {Movie} from '@src/model';
import {IMoviesQueryResult} from '@src/interfaces';

import {useDependenciesContainer} from '@src/DI/container';
import {useInjectedDI} from '@src/DI/Context';
import {initialEnvRegister} from '@src/DI/initialEnvRegister';

function App(): React.JSX.Element {
  initialEnvRegister();
  const deps = useInjectedDI(useDependenciesContainer());

  const {data, isLoading, isSuccess, isError, error}: IMoviesQueryResult =
    deps.moviesQueryResult();
  const movieList: Movie[] = data?.results || [];

  let content: React.JSX.Element | null;

  switch (true) {
    case isLoading:
      content = <ActivityIndicator />;
      break;
    case isSuccess:
      content = <MovieList movieList={movieList} />;
      break;
    case isError:
      content = <Text>{error?.toString()}</Text>;
      break;
    default:
      content = null;
  }

  return (
    <SafeAreaView style={Styles.safeAreaView}>
      <View style={Styles.container}>{content}</View>
    </SafeAreaView>
  );
}

export default App;
