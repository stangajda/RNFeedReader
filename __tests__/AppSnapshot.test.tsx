import React, {ReactNode} from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import App from '@src/App';
import {Provider} from 'react-redux';
import {store} from '@src/Store';
import {useGetMoviesQuery} from '@src/ApiSlice';
import {Movies} from '@src/Model';
import {FetchBaseQueryError} from '@reduxjs/toolkit/query';
import {SerializedError} from '@reduxjs/toolkit';

function wrapper({children}: {children: ReactNode}) {
  return <Provider store={store}>{children}</Provider>;
}

interface MoviesQueryResult {
  data?: Movies;
  isLoading: boolean;
  isSuccess: boolean;
  isError: boolean;
  error: FetchBaseQueryError | SerializedError | undefined;
}

jest.mock('@src/apiSlice', () => {
  const originalModule = jest.requireActual('@src/apiSlice');
  return {
    __esModule: true,
    ...originalModule,
    useGetMoviesQuery: jest.fn<
      ReturnType<typeof useGetMoviesQuery>,
      Parameters<typeof useGetMoviesQuery>
    >(),
  };
});

const mockUseGetMoviesQuery = useGetMoviesQuery as jest.Mock<MoviesQueryResult>;

describe('check movies list view to match recorded snapshot', () => {
  describe('when movies list is loaded', () => {
    beforeAll(() => {
      const dataResult: Movies = require('./StubMovieListResponseResult.json');
      mockUseGetMoviesQuery.mockReturnValueOnce({
        data: dataResult,
        isLoading: false,
        isSuccess: true,
        isError: false,
        error: undefined,
      });
    });
    it('it should match movie list loaded image json', () => {
      const tree = renderer.create(wrapper({children: <App />})).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when movies list is loading', () => {
    beforeAll(() => {
      mockUseGetMoviesQuery.mockReturnValueOnce({
        isLoading: true,
        isSuccess: false,
        isError: false,
        error: undefined,
      });
    });
    it('it should match movie list loading image json', () => {
      const tree = renderer.create(wrapper({children: <App />})).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when movies list is error', () => {
    beforeAll(() => {
      mockUseGetMoviesQuery.mockReturnValueOnce({
        isLoading: false,
        isSuccess: false,
        isError: true,
        error: new Error('stub error message'),
      });
    });
    it('it should match movie list error image json', () => {
      const tree = renderer.create(wrapper({children: <App />})).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when movies list is empty', () => {
    beforeAll(() => {
      mockUseGetMoviesQuery.mockReturnValueOnce({
        isLoading: false,
        isSuccess: true,
        isError: false,
        error: undefined,
      });
    });
    it('it should match movie list empty image json', () => {
      const tree = renderer.create(wrapper({children: <App />})).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('when loading success and error at the same time return false', () => {
    beforeAll(() => {
      mockUseGetMoviesQuery.mockReturnValueOnce({
        isLoading: false,
        isSuccess: false,
        isError: false,
        error: undefined,
      });
    });
    it('it should match movie list empty image json', () => {
      const tree = renderer.create(wrapper({children: <App />})).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
