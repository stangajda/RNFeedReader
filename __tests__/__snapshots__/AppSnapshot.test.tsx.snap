// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check movies list view to match recorded snapshot when loading success and error at the same time return false it should match movie list empty image json 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "padding": 24,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`check movies list view to match recorded snapshot when movies list is empty it should match movie list empty image json 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "padding": 24,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`check movies list view to match recorded snapshot when movies list is error it should match movie list loading image json 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "padding": 24,
      }
    }
  >
    <Text>
      Error: stub error message
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`check movies list view to match recorded snapshot when movies list is loaded should match movie list loaded image json 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "padding": 24,
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": 1,
            "poster_path": "0-poster_path",
            "title": "mock-0-title",
            "vote_average": 6.1,
            "vote_count": 1724,
          },
          {
            "id": 2,
            "poster_path": "1-poster_path",
            "title": "mock-1-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 3,
            "poster_path": "2-poster_path",
            "title": "mock-2-title",
            "vote_average": 6.9,
            "vote_count": 711,
          },
          {
            "id": 4,
            "poster_path": "3-poster_path",
            "title": "mock-3-title",
            "vote_average": 6.5,
            "vote_count": 1724,
          },
          {
            "id": 5,
            "poster_path": "4-poster_path",
            "title": "mock-4-title",
            "vote_average": 6.5,
            "vote_count": 1724,
          },
          {
            "id": 6,
            "poster_path": "5-poster_path",
            "title": "mock-5-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 7,
            "poster_path": "6-poster_path",
            "title": "mock-6-title",
            "vote_average": 6.5,
            "vote_count": 1724,
          },
          {
            "id": 8,
            "poster_path": "7-poster_path",
            "title": "mock-7-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 9,
            "poster_path": "8-poster_path",
            "title": "mock-8-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 10,
            "poster_path": "9-poster_path",
            "title": "mock-9-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 11,
            "poster_path": "10-poster_path",
            "title": "mock-10-title",
            "vote_average": 7.1,
            "vote_count": 711,
          },
          {
            "id": 12,
            "poster_path": "11-poster_path",
            "title": "mock-11-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 13,
            "poster_path": "12-poster_path",
            "title": "mock-12-title",
            "vote_average": 7.2,
            "vote_count": 711,
          },
          {
            "id": 14,
            "poster_path": "13-poster_path",
            "title": "mock-13-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 15,
            "poster_path": "14-poster_path",
            "title": "mock-14-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 16,
            "poster_path": "15-poster_path",
            "title": "mock-15-title",
            "vote_average": 6.5,
            "vote_count": 675,
          },
          {
            "id": 17,
            "poster_path": "16-poster_path",
            "title": "mock-16-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 18,
            "poster_path": "17-poster_path",
            "title": "mock-17-title",
            "vote_average": 6.5,
            "vote_count": 654,
          },
          {
            "id": 19,
            "poster_path": "18-poster_path",
            "title": "mock-18-title",
            "vote_average": 6.2,
            "vote_count": 711,
          },
          {
            "id": 20,
            "poster_path": "19-poster_path",
            "title": "mock-19-title",
            "vote_average": 6.5,
            "vote_count": 711,
          },
          {
            "id": 21,
            "poster_path": "20-poster_path",
            "title": "mock-20-title",
            "vote_average": 4.4,
            "vote_count": 3212,
          },
          {
            "id": 22,
            "poster_path": "21-poster_path",
            "title": "mock-21-title",
            "vote_average": 9.1,
            "vote_count": 765,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2000-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-0-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.1
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    1724
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2001-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-1-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2002-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-2-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.9
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2003-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-3-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    1724
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2004-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-4-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    1724
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2005-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-5-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2006-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-6-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    1724
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2007-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-7-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2008-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-8-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "flexDirection": "row",
                "padding": 8,
              }
            }
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w2009-poster_path",
                }
              }
              style={
                {
                  "height": 88,
                  "marginRight": 8,
                  "width": 64,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View>
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={2}
                  style={
                    {
                      "color": "black",
                      "flexShrink": 1,
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  mock-9-title
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#444",
                        "fontWeight": "bold",
                      }
                    }
                  >
                    6.5
                     
                  </Text>
                  <Text
                    style={
                      {
                        "color": "grey",
                      }
                    }
                  >
                    (
                    711
                    )
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`check movies list view to match recorded snapshot when movies list is loading it should match movie list loading image json 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "padding": 24,
      }
    }
  >
    <ActivityIndicator />
  </View>
</RCTSafeAreaView>
`;
